cpp/src/groupby/sort/group_scan_util.cuh

```diff
@@ -129,12 +129,12 @@ struct group_scan_functor<K, T, std::enable_if_t<is_group_scan_supported<K, T>()
     if (values.has_nulls()) {
       auto input = thrust::make_transform_iterator(
         make_null_replacement_iterator(*values_view, OpType::template identity<DeviceType>()),
-        thrust::identity<ResultDeviceType>{});
+        cudf::detail::cast_fn<ResultDeviceType>{});
```

Comment:
ELI5:  What would happen if we used `cuda::std::identity<ResultDeviceType>{}`?  I'm not sure I've appreciated why we needed to introduce `cast_fn<>`.

Reply:
Unlike `thrust::identity<T>` which can be used as type converter since it takes a template parameter, `cuda::std::identity` / `std::identity` is not a template callable per https://en.cppreference.com/w/cpp/utility/functional/identity

Reply:
Oh, right.  I see your point. Thank you.